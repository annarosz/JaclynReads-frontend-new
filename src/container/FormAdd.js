import React from "react";
import axios from "axios";
import FormInput from "../components/FormInput";
import FormTextArea from "../components/FormTextArea";
import FormCheckbox from "../components/FormCheckbox";
import Button from "../components/Button";
import "../container/FormAdd.css";

class FormAdd extends React.Component {
  state = {
    newReview: {
      title: this.props.title || "",
      author: this.props.author || "",
      review: this.props.review || "",
      publisher: this.props.publisher || "",
      yearPublished: this.props.yearPublished || "2019",
      genre: this.props.genre || [],
      isbn: this.props.isbn || "",
      linkToBuy: this.props.linkToBuy || "",
      topPick: this.props.topPick || false,
      seoKeyword: this.props.seoKeyword || "",
      url: this.props.url || ""
    },
    image: "",

    genreOptions: [
      "Australian Fiction",
      "Contemporary Fiction",
      "Crime",
      "Essays",
      "Graphic Novels",
      "In Translation",
      "Memoir",
      "Narrative Non-Fiction",
      "Short Stories",
      "Young Adult(YA)"
    ],
    topPickOptions: ["Top Pick"],
    errors: {
      title: "",
      author: "",
      publisher: "",
      yearPublished: "",
      isbn: "",
      linkToBuy: "",
      review: "",
      genre: "",
  }
}

  handleTextArea = e => {
    e.preventDefault();
    let errors = this.state.errors;
    let value = e.target.value;

    errors.review = value.length < 2 ? "Must enter a review" : "";

    this.setState(prevState => ({
      newReview: {
        ...prevState.newReview,
        review: value
      }
    }));
  };

  handleInput = e => {
    e.preventDefault();
    let errors = this.state.errors;
    let value = e.target.value;
    let name = e.target.name;

    switch (name) {
      case "title":
        errors.title = value.length < 2 ? "Must enter a title" : "";
        break;
      case "author":
        errors.author = value.length < 2 ? "Must enter an author" : "";
        break;
      case "publisher":
        errors.publisher = value.length < 2 ? "Must enter an publisher" : "";
        break;
      case "yearPublished":
        errors.yearPublished = value > 1900 ? "Must enter year" : "";
        break;
      case "isbn":
        errors.isbn = value.length < 2 ? "Must enter ISBN" : "";
        break;
      case "linkToBuy":
        errors.linkToBuy = value.length < 2 ? "Must enter a link" : "";
        break;
      default:
        break;
    }

    this.setState({ errors, [name]: value }, () => {});

    this.setState(prevState => ({
      newReview: {
        ...prevState.newReview,
        [name]: value
      }
    }));
  };

  handleInputSeo = e => {
    let value = e.target.value;
    let name = e.target.name;
    const split = value.split(",");

    this.setState(prevState => ({
      newReview: {
        ...prevState.newReview,
        [name]: split
      }
    }));
  };

  handleCheckBox = e => {
    const newSelection = e.target.value;
    let newSelectionArray;

    if (this.state.newReview.genre.indexOf(newSelection) > -1) {
      newSelectionArray = this.state.newReview.genre.filter(
        s => s !== newSelection
      );
    } else {
      newSelectionArray = [...this.state.newReview.genre, newSelection];
    }

    let errors = this.state.errors;
    errors.genre = newSelectionArray.length < 1 ? "Must enter a genre" : "";

    this.setState(prevState => ({
      newReview: { ...prevState.newReview, genre: newSelectionArray }
    }));
  };

  handleCheckbox = e => {
    if (e.target.checked === true) {
      this.setState(prevState => ({
        newReview: {
          ...prevState.newReview,
          topPick: true
        }
      }));
    } else {
      this.setState(prevState => ({
        newReview: {
          ...prevState.newReview,
          topPick: false
        }
      }));
    }
  };

  handleYear = e => {
    let value = e.target.value;
    this.setState(prevState => ({
      newReview: {
        ...prevState.newReview,
        yearPublished: value
      }
    }));
  };

  handleClearForm = e => {
    e.preventDefault();
    this.setState({
      newReview: {
        title: "",
        author: "",
        review: "",
        publisher: "",
        yearPublished: "",
        genre: [],
        isbn: "",
        linkToBuy: "",
        topPick: "",
        seoKeyword: ""
      }
    });
  };

  validateForm = errors => {
    let valid = true;
    Object.values(errors).forEach(val => val.length > 0 && (valid = false));
    return valid;
  };

  handleUploadImage = e => {
    e.preventDefault();
    const file = e.target.parentNode.childNodes[10].files[0];
    this.setState({ image: file });
  };

  handleFormSubmit = e => {
    e.preventDefault();

    if (this.validateForm(this.state.errors)) {
    } else {
    }
    const newReview = this.state.newReview;
    const file = this.state.image;
    const id = this.props.id;
    const data = { id: id, newReview: newReview };

    if (window.location.pathname.split("/").pop() === "adminshow") {
      const stringifyData = JSON.stringify(data);
      const formData = new FormData();
      formData.append("data", stringifyData);
      formData.append("file", file);

      const token = localStorage.getItem("token");
      const headers = { token: token };

      axios
        .put(
          `${process.env.REACT_APP_API_URL}/protected/updateReview`,
          formData,
          { headers }
        )
        .then(res => {
          alert("review updated");
          this.props.refresh(res);
        })
        .catch(err => {});
    } else {
      const stringifyData = JSON.stringify(newReview);
      const formData = new FormData();
      formData.append("data", stringifyData);
      formData.append("file", file);

      const token = localStorage.getItem("token");
      const headers = { token: token };

      axios
        .post(
          `${process.env.REACT_APP_API_URL}/protected/createReview`,
          formData,
          { headers }
        )
        .then(res => {
          alert("Review Saved");
          this.props.history.push("/auth/adminshow");
        })
        .catch(err => {});
    }
  };

  handleCancelForm = e => {
    e.preventDefault();
    if (window.location.pathname === "/auth/adminshow") {
      return window.location.reload();
    } else {
      return this.props.history.push("/auth/adminshow");
    }
  };

  render() {
    const { errors } = this.state;

    return (
      <>
        <div className="formContainer">
          <form className="form-add" encType="multipart/form-data">
            <FormInput
              inputType={"text"}
              title={"Book Title"}
              name={"title"}
              value={this.state.newReview.title}
              placeholder={"Enter book name"}
              handleChange={this.handleInput}
            />{" "}
            {errors.title.length > 0 && (
              <span className="error">{errors.title}</span>
            )}
            <FormInput
              inputType={"text"}
              title={"Author"}
              name={"author"}
              value={this.state.newReview.author}
              placeholder={"Enter author name"}
              handleChange={this.handleInput}
            />{" "}
            {errors.author.length > 0 && (
              <span className="error">{errors.author}</span>
            )}
            <FormTextArea
              title={"Review"}
              rows={10}
              value={this.state.newReview.review}
              name={"review"}
              handleChange={this.handleTextArea}
              placeholder={"Enter your review"}
            />
            {errors.review.length > 0 && (
              <span className="error">{errors.review}</span>
            )}
            <FormInput
              inputType={"text"}
              title={"Publisher"}
              name={"publisher"}
              value={this.state.newReview.publisher}
              placeholder={"Enter publisher"}
              handleChange={this.handleInput}
            />{" "}
            {errors.publisher.length > 0 && (
              <span className="error">{errors.publisher}</span>
            )}
            <FormInput
              inputType={"number"}
              name={"yearPublished"}
              title={"Year Published"}
              value={this.state.newReview.yearPublished}
              placeholder={"Published Year"}
              handleChange={this.handleYear}
            />{" "}
            {errors.yearPublished.length > 0 && (
              <span className="error">{errors.yearPublished}</span>
            )}
            <label htmlFor="image-upload">Upload Image</label>
            <input
              type="file"
              name="image-upload"
              id="image-load"
              onChange={this.handleUploadImage}
            />
            {this.state.newReview.url && (
              <div>
                <input
                  className="form-control"
                  type="text"
                  value={this.state.newReview.url}
                  readOnly
                />
              </div>
            )}
            <FormCheckbox
              title={"Genre"}
              name={"genre"}
              options={this.state.genreOptions}
              selectedOptions={this.state.newReview.genre}
              handleChange={this.handleCheckBox}
            />{" "}
            {errors.genre.length > 0 && (
              <span className="error">{errors.genre}</span>
            )}
            <FormInput
              inputType={"text"}
              title={"ISBN"}
              name={"isbn"}
              value={this.state.newReview.isbn}
              placeholder={"Enter ISBN"}
              handleChange={this.handleInput}
            />{" "}
            {errors.isbn.length > 0 && (
              <span className="error">{errors.isbn}</span>
            )}
            <FormInput
              inputType={"text"}
              title={"Link to Buy"}
              name={"linkToBuy"}
              value={this.state.newReview.linkToBuy}
              placeholder={"Enter URL"}
              handleChange={this.handleInput}
            />{" "}
            {errors.linkToBuy.length > 0 && (
              <span className="error">{errors.linkToBuy}</span>
            )}
            <div className="form-group">
              <label htmlFor="check">Top Pick</label>
              <input
                onChange={this.handleCheckbox}
                type="checkbox"
                id="checkbox"
                name="topPick"
                checked={this.state.newReview.topPick}
              />
            </div>
            <FormInput
              inputType={"text"}
              title={"Seo Keywords (Separate by comma)"}
              name={"seoKeyword"}
              value={this.state.newReview.seoKeyword}
              placeholder={"Enter SEO Keywords"}
              handleChange={this.handleInputSeo}
            />{" "}
            <Button
              action={this.handleFormSubmit}
              type={"primary"}
              title={"Submit"}
            />{" "}
            {/*Submit */}
            <Button
              action={this.handleClearForm}
              type={"secondary"}
              title={"Clear"}
              // style={buttonStyle}
            />{" "}
            <Button
              action={this.handleCancelForm}
              type={"secondary"}
              title={"Cancel"}
            />{" "}
          </form>
        </div>
      </>
    );
  }
}

export default FormAdd;
