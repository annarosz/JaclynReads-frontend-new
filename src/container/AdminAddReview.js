import React from "react";
import IntroSectionFlexi from "../components/IntroSectionFlexi";
import FormAdd from "../container/FormAdd";

class AdminAddReview extends React.Component {

  render() {

    return (
      <div className="containerSecondary" style={{ backgroundColor: '#E7FBAD' }}>
        <IntroSectionFlexi
          headingOne="Create a new review? You got it!" />
        <FormAdd {...this.props}/>
      </div>
    );
  }
}

export default AdminAddReview;