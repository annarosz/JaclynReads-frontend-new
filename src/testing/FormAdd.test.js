import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";
import AdminAddReview from "../container/AdminAddReview";
import FormAdd from "../container/FormAdd";

describe("<FormAdd />", () => {
  const reviewList = [
    {
      name: "Three Little Pigs",
      author: "Big Bad Wolf",
      id: "5d19cd74a0c3e2c7f68ff800",
      inputType: "type",
      value: "value",
      handleChange: "handleChange"
    }
  ];

  it("displays the form", () => {
    const review = [];
    const wrapper = shallow(<FormAdd />);
    expect(wrapper).toMatchSnapshot();
  });

  it("fields are rendered", () => {
    const wrapper = render(<FormAdd />);
    expect(wrapper).toMatchSnapshot();
  });

});
