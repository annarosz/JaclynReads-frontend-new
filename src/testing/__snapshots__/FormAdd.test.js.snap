// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormAdd /> displays the form 1`] = `
<Fragment>
  <div
    className="formContainer"
  >
    <form
      className="form-add"
      encType="multipart/form-data"
    >
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="title"
        placeholder="Enter book name"
        title="Book Title"
        value=""
      />
       
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="author"
        placeholder="Enter author name"
        title="Author"
        value=""
      />
       
      <TextArea
        handleChange={[Function]}
        name="review"
        placeholder="Enter your review"
        rows={10}
        title="Review"
        value=""
      />
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="publisher"
        placeholder="Enter publisher"
        title="Publisher"
        value=""
      />
       
      <FormInput
        handleChange={[Function]}
        inputType="number"
        name="yearPublished"
        placeholder="Published Year"
        title="Year Published"
        value="2019"
      />
       
      <label
        htmlFor="image-upload"
      >
        Upload Image
      </label>
      <input
        id="image-load"
        name="image-upload"
        onChange={[Function]}
        type="file"
      />
      <FormCheckbox
        handleChange={[Function]}
        name="genre"
        options={
          Array [
            "Australian Fiction",
            "Contemporary Fiction",
            "Crime",
            "Essays",
            "Graphic Novels",
            "In Translation",
            "Memoir",
            "Narrative Non-Fiction",
            "Short Stories",
            "Young Adult(YA)",
          ]
        }
        selectedOptions={Array []}
        title="Genre"
      />
       
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="isbn"
        placeholder="Enter ISBN"
        title="ISBN"
        value=""
      />
       
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="linkToBuy"
        placeholder="Enter URL"
        title="Link to Buy"
        value=""
      />
       
      <div
        className="form-group"
      >
        <label
          htmlFor="check"
        >
          Top Pick
        </label>
        <input
          checked={false}
          id="checkbox"
          name="topPick"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <FormInput
        handleChange={[Function]}
        inputType="text"
        name="seoKeyword"
        placeholder="Enter SEO Keywords"
        title="Seo Keywords (Separate by comma)"
        value=""
      />
       
      <Button
        action={[Function]}
        title="Submit"
        type="primary"
      />
       
      <Button
        action={[Function]}
        title="Clear"
        type="secondary"
      />
       
      <Button
        action={[Function]}
        title="Cancel"
        type="secondary"
      />
       
    </form>
  </div>
</Fragment>
`;

exports[`<FormAdd /> fields are mounted 1`] = `
<div
  className="formContainer"
>
  <form
    className="form-add"
    encType="multipart/form-data"
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="title"
      >
        Book Title
      </label>
      <input
        className="form-control"
        id="title"
        min="1999"
        name="title"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter book name"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="author"
      >
        Author
      </label>
      <input
        className="form-control"
        id="author"
        min="1999"
        name="author"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter author name"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Review
      </label>
      <textarea
        className="form-control"
        name="review"
        onChange={[Function]}
        placeholder="Enter your review"
        rows={10}
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="publisher"
      >
        Publisher
      </label>
      <input
        className="form-control"
        id="publisher"
        min="1999"
        name="publisher"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter publisher"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="yearPublished"
      >
        Year Published
      </label>
      <input
        className="form-control"
        id="yearPublished"
        min="1999"
        name="yearPublished"
        noValidate={true}
        onChange={[Function]}
        placeholder="Published Year"
        type="number"
        value="2019"
      />
    </div>
     
    <label
      htmlFor="image-upload"
    >
      Upload Image
    </label>
    <input
      id="image-load"
      name="image-upload"
      onChange={[Function]}
      type="file"
    />
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="genre"
      >
        Genre
      </label>
      <div
        className="checkbox"
      >
        <label
          className="checkbox-inline"
          key="Australian Fiction"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Australian Fiction"
          />
          Australian Fiction
        </label>
        <label
          className="checkbox-inline"
          key="Contemporary Fiction"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Contemporary Fiction"
          />
          Contemporary Fiction
        </label>
        <label
          className="checkbox-inline"
          key="Crime"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Crime"
          />
          Crime
        </label>
        <label
          className="checkbox-inline"
          key="Essays"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Essays"
          />
          Essays
        </label>
        <label
          className="checkbox-inline"
          key="Graphic Novels"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Graphic Novels"
          />
          Graphic Novels
        </label>
        <label
          className="checkbox-inline"
          key="In Translation"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="In Translation"
          />
          In Translation
        </label>
        <label
          className="checkbox-inline"
          key="Memoir"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Memoir"
          />
          Memoir
        </label>
        <label
          className="checkbox-inline"
          key="Narrative Non-Fiction"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Narrative Non-Fiction"
          />
          Narrative Non-Fiction
        </label>
        <label
          className="checkbox-inline"
          key="Short Stories"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Short Stories"
          />
          Short Stories
        </label>
        <label
          className="checkbox-inline"
          key="Young Adult(YA)"
        >
          <input
            checked={false}
            id="genre"
            name="genre"
            noValidate={true}
            onChange={[Function]}
            type="checkbox"
            value="Young Adult(YA)"
          />
          Young Adult(YA)
        </label>
      </div>
    </div>
     
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="isbn"
      >
        ISBN
      </label>
      <input
        className="form-control"
        id="isbn"
        min="1999"
        name="isbn"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter ISBN"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="linkToBuy"
      >
        Link to Buy
      </label>
      <input
        className="form-control"
        id="linkToBuy"
        min="1999"
        name="linkToBuy"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter URL"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="form-group"
    >
      <label
        htmlFor="check"
      >
        Top Pick
      </label>
      <input
        checked={false}
        id="checkbox"
        name="topPick"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="seoKeyword"
      >
        Seo Keywords (Separate by comma)
      </label>
      <input
        className="form-control"
        id="seoKeyword"
        min="1999"
        name="seoKeyword"
        noValidate={true}
        onChange={[Function]}
        placeholder="Enter SEO Keywords"
        type="text"
        value=""
      />
    </div>
     
    <div
      className="btn btn-primary"
      onClick={[Function]}
    >
      Submit
    </div>
     
    <div
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Clear
    </div>
     
    <div
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Cancel
    </div>
     
  </form>
</div>
`;

exports[`<FormAdd /> fields are rendered 1`] = `
<div
  class="formContainer"
>
  <form
    class="form-add"
    enctype="multipart/form-data"
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="title"
      >
        Book Title
      </label>
      <input
        class="form-control"
        id="title"
        min="1999"
        name="title"
        novalidate=""
        placeholder="Enter book name"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="author"
      >
        Author
      </label>
      <input
        class="form-control"
        id="author"
        min="1999"
        name="author"
        novalidate=""
        placeholder="Enter author name"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="form-group"
    >
      <label
        class="form-label"
      >
        Review
      </label>
      <textarea
        class="form-control"
        name="review"
        placeholder="Enter your review"
        rows="10"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="publisher"
      >
        Publisher
      </label>
      <input
        class="form-control"
        id="publisher"
        min="1999"
        name="publisher"
        novalidate=""
        placeholder="Enter publisher"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="yearPublished"
      >
        Year Published
      </label>
      <input
        class="form-control"
        id="yearPublished"
        min="1999"
        name="yearPublished"
        novalidate=""
        placeholder="Published Year"
        type="number"
        value="2019"
      />
    </div>
     
    <label
      for="image-upload"
    >
      Upload Image
    </label>
    <input
      id="image-load"
      name="image-upload"
      type="file"
    />
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="genre"
      >
        Genre
      </label>
      <div
        class="checkbox"
      >
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Australian Fiction"
          />
          Australian Fiction
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Contemporary Fiction"
          />
          Contemporary Fiction
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Crime"
          />
          Crime
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Essays"
          />
          Essays
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Graphic Novels"
          />
          Graphic Novels
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="In Translation"
          />
          In Translation
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Memoir"
          />
          Memoir
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Narrative Non-Fiction"
          />
          Narrative Non-Fiction
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Short Stories"
          />
          Short Stories
        </label>
        <label
          class="checkbox-inline"
        >
          <input
            id="genre"
            name="genre"
            novalidate=""
            type="checkbox"
            value="Young Adult(YA)"
          />
          Young Adult(YA)
        </label>
      </div>
    </div>
     
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="isbn"
      >
        ISBN
      </label>
      <input
        class="form-control"
        id="isbn"
        min="1999"
        name="isbn"
        novalidate=""
        placeholder="Enter ISBN"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="linkToBuy"
      >
        Link to Buy
      </label>
      <input
        class="form-control"
        id="linkToBuy"
        min="1999"
        name="linkToBuy"
        novalidate=""
        placeholder="Enter URL"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="form-group"
    >
      <label
        for="check"
      >
        Top Pick
      </label>
      <input
        id="checkbox"
        name="topPick"
        type="checkbox"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="seoKeyword"
      >
        Seo Keywords (Separate by comma)
      </label>
      <input
        class="form-control"
        id="seoKeyword"
        min="1999"
        name="seoKeyword"
        novalidate=""
        placeholder="Enter SEO Keywords"
        type="text"
        value=""
      />
    </div>
     
    <div
      class="btn btn-primary"
    >
      Submit
    </div>
     
    <div
      class="btn btn-secondary"
    >
      Clear
    </div>
     
    <div
      class="btn btn-secondary"
    >
      Cancel
    </div>
     
  </form>
</div>
`;
